utopia-remix/app/routes/v1.project.$id.tsx

```diff
@@ -1,14 +1,40 @@
 import { ActionFunctionArgs, LoaderFunctionArgs } from '@remix-run/node'
 import { proxy } from '../util/proxy.server'
-import { handle, handleOptions } from '../util/api.server'
+import { ensure, handle, handleOptions, requireUser } from '../util/api.server'
+import { Params } from '@remix-run/react'
+import { Status } from '../util/statusCodes'
+import { hasUserProjectPermission } from '../services/permissionsService.server'
+import { UserProjectPermission } from '../types'
 
 export async function loader(args: LoaderFunctionArgs) {
   return handle(args, {
     OPTIONS: handleOptions,
-    GET: proxy,
+    GET: getProject,
   })
 }
 
+export async function getProject(req: Request, params: Params<string>) {
+  const { id: projectId } = params
+  ensure(projectId != null, 'project is null', Status.BAD_REQUEST)
+  let userId
```

Comment:
I think you can safely add a separate function that is the non-require variant of `requireUser` (which basically just does `getUserFromSessionId`) and use the `user|null` return. Moreover I don't understand if `ANON` is actually useful or not, if not I think it makes more sense to just drop it and rely on the user being null or not

Reply:
I also don't think this is resolved, see also @Rheeseyb 's comment about `ANON`